packages/webapp/src/_app/hooks/box-queries.ts

```diff
@@ -18,3 +20,109 @@ export const useElectionStatus = () =>
+    account?: string
+): Query<CurrentMemberElectionVotingDataQuery> => {
+    const query = useQuery<any>(
+        account
+            ? `
+    {
+      members(le: "${account}", ge: "${account}") {
+        edges {
+          node {
+            account
+            elections(last: 1) {
+              edges {
+                node {
+                  time
+                  votes {
+                    edges {
+                      node {
+                        group {
+                          round {
+                            round
+                            votingBegin
+                            votingEnd
+                            votingStarted
+                            votingFinished
+                            resultsAvailable
+                          }
+                          winner {
+                            account
+                            profile {
+                              name
+                            }
+                          }
+                        }
+                        candidate {
+                          account
+                          profile {
+                            name
+                          }
+                        }
+                        video
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    `
+            : ""
+    );
+
+    if (query.data) {
+        const lastElection =
+            query.data.members.edges?.[0]?.node.elections?.edges?.[0].node;
+        if (lastElection) {
+            query.data.electionTime = dayjs(lastElection.time);
+            query.data.votes =
+                lastElection.votes?.edges?.map((voteEdge: any) => ({
+                    roundIndex: voteEdge.node.group.round.round,
+                    votingBegin: dayjs(voteEdge.node.group.round.votingBegin),
+                    votingEnd: dayjs(voteEdge.node.group.round.votingEnd),
+                    votingStarted: voteEdge.node.group.round.votingStarted,
+                    votingFinished: voteEdge.node.group.round.votingFinished,
+                    resultsAvailable:
+                        voteEdge.node.group.round.resultsAvailable,
+                    candidate: voteEdge.node.candidate
+                        ? {
+                              account: voteEdge.node.candidate.account,
+                              name: voteEdge.node.candidate.profile.name,
+                          }
+                        : undefined,
+                    winner: voteEdge.node.group.winner
+                        ? {
+                              account: voteEdge.node.group.winner.account,
+                              name: voteEdge.node.group.winner.profile.name,
+                          }
+                        : undefined,
+                    video: voteEdge.node.video,
+                })) || [];
+        }
+    }
```

Comment:
question: any suggestions on how to make this function more elegant? I see us repeating the pattern of massaging/formatting retrieved box queries... Question is, how to set an elegant pattern so we don't grow this kind of code?

Reply:
That's a good question. I'll give it some thought.

suggestion: this is just a thought--I'm not totally committed to it--but our queries file for react-query queries became super long because it was all in one file in `_app`. With our new box queries, should we consider putting them inside the directories their related to? So instead of this query going in `_app/hooks/box-queries.ts`, it would be in `election/hooks/box-queries.ts`? Of course, any common queries could still go in the former.

Reply:
I want to say something like tagged templates. If we could store the query in a string (get them out of this code), and delay the interpolation, that would be much cleaner.
The way them seem structured makes it doable, but with some hacking...
https://betterprogramming.pub/javascript-how-backticks-work-de269e0fb8ba
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
I like both suggestions. Since it's about just splitting and moving around I'm going to keep like this in this PR and we can add these improvements to our backlog!